src/visualization/parsers/perceivedLoadTimeLayout.ts

```diff
@@ -0,0 +1,185 @@
+/// <reference path="../../../node_modules/clarity-js/clarity.d.ts" />
+import { IParser } from "../components/Snapshot";
+import { IAttributes, ILayoutState, IElementLayoutState, IDoctypeLayoutState, ITextLayoutState, IIgnoreLayoutState, Action } from "clarity-js/clarity";
+import { IPerceivedElementLayoutState } from "./IPerceivedElementLayoutState";
+import { Layout } from  "../parsers/layout";
+
+export default class PerceivedLoadTimeLayout extends Layout {    
+    
+    setup(document: Document, frame: HTMLIFrameElement, base: string, thumbnail? : boolean) {
+        this.layouts = {};
+        this.document = document;
+        this.base = base;
+    }
+
+    private insertPerceivedLayout (layoutState: ILayoutState) {
+        var doc = this.document;
+        var state: any = layoutState as IPerceivedElementLayoutState;
+        var parent = this.layouts[state.parent];
+        var next = state.next in this.layouts ? this.layouts[state.next] : null;
+        switch (state.tag) {
```

Comment:
>switch (state.tag) { [](start = 8, length = 20)

huge parts of this file are still copy pasted from layout.ts. Can we make a protected helper method which is a noop in layout.ts but does your small bit of extra work for the child class?